web/src/components/VideoPlayer.jsx

```diff
@@ -0,0 +1,44 @@
+import { h, Component } from 'preact';
+import videojs from 'video.js';
+import 'videojs-playlist';
+import 'video.js/dist/video-js.css';
+
+const defaultOptions = {
+  controls: true,
+  fluid: true,
+};
+
+export default class VideoPlayer extends Component {
```

Comment:
Please use functional components and not class components

Reply:
This is the last class component. Any help on how I would convert it to a functional component would be appreciated.

Reply:
I tried to convert this to a functional component with the code below, but get the following error when navigating:
![image](https://user-images.githubusercontent.com/747670/120557310-b78a3e80-c3cb-11eb-89f9-7df789d6662c.png)

```javascript
export default function VideoPlayer({ children, options, onReady = () => {} }) {
  const playerRef = useRef(null);
  useEffect(() => {
    if (playerRef.current) {
      const player = videojs(playerRef.current, { ...defaultOptions, ...options }, () => {
        onReady(player);
      });
      return () => {
        player.dispose();
      };
    }
  }, [options, onReady]);

  return (
    <div data-vjs-player>
      <video ref={playerRef} className="video-js vjs-default-skin" controls playsInline />
      {children}
    </div>
  );
}
```

Maybe it's related to https://github.com/preactjs/preact/issues/2341#issuecomment-585816891?